libc/src/time/strftime_core/num_converter.h

```diff
@@ -0,0 +1,208 @@
+//===-- Format specifier converter for printf -------------------*- C++ -*-===//
```

Comment:
I'd recommend reading through the printf `int_converter`: https://github.com/llvm/llvm-project/blob/main/libc/src/stdio/printf_core/int_converter.h

It does very similar things to this file, and you may want to use similar techniques.